client/signup/steps/site-style/index.jsx

```diff
@@ -54,7 +54,9 @@ export class SiteStyleStep extends Component {
 
 	handleSubmit = event => {
 		event.preventDefault();
-		this.props.submitSiteStyle( this.props.siteStyle, this.getSelectedStyleDataById().theme );
+		const selectedStyle =
+			this.getSelectedStyleDataById() || this.getSelectedStyleDataById( 'default' );
+		this.props.submitSiteStyle( selectedStyle.id, selectedStyle.theme );
```

Comment:
Thanks for doing this! I was meaning to fix it in the next iteration, but, ah... reasons :trollface: 

What do you think about returning a default from `getSelectedStyleDataById()` instead? That way we can cover other usages of this method for the case that `this.props.siteStyle` isn't set.

And rather than specify `'default'` we should probably just reference the first item in the `this.props.styleOptions` array. Two reasons why:

1. We don't know if there will be a style with an id of `'default'` for all style variations (and we won't have to enforce this arbitrary rule).
2. It keeps it consistent with the way we [render the radio option fields](https://github.com/Automattic/wp-calypso/blob/cd93dc3aa2743dc777e58f53bc622df26ed79f7b/client/signup/steps/site-style/index.jsx#L70) (where the default selected input is the first)

For example:

```
    // if we don't get anything back from `styleOptions` using an `id`, just grab the first one.
	getSelectedStyleDataById( id ) {
		return find( this.props.styleOptions, [ 'id', id || this.props.siteStyle ] ) || this.props.styleOptions[ 0 ];
	}
```

Then a new unit test might be in order:

```
	test( 'should select and submit the first item from the options array if no site style yet saved in state', () => {
		const wrapper = shallow( <SiteStyleStep { ...defaultProps } siteStyle={ null } /> );
		const siteStyleContent = shallow( wrapper.instance().renderContent() );
		const firstInputField = siteStyleContent.find( 'FormRadio' ).get( 0 );
        // check that the default is the first item rendered
		expect( firstInputField.props.value ).toEqual( defaultProps.styleOptions[ 0 ].id );
         // check that it's been selected
		expect( firstInputField.hasClass( 'is-checked' ) ).toBe( true );

		wrapper.instance().handleSubmit( {
			preventDefault: () => {},
		} );
        // check that we pass the default site option onSubmit
		expect( defaultProps.submitSiteStyle ).toHaveBeenCalledWith(
			defaultProps.styleOptions[ 0 ].id,
			defaultProps.styleOptions[ 0 ].theme,
			defaultProps.styleOptions[ 0 ].label
		);
	} );
```
Disclaimer: I didn't mean to write the code for this PR, the above code was lurking in my next commit for #29863 :)

Reply:
> With a default return value, could the caller differentiate between an exact match for the id value and a default value

You're making sense! I didn't consider that. Feel free to ignore my comment and dismiss the review after the other changes in are (if you decide to go with them) 👍

Reply:
Addressed feedback with changes pushed in [9b9c364](9b9c364).